libflux/src/core/parser/mod.rs

```diff
@@ -590,6 +591,38 @@ impl Parser {
         }));
     }
 
+    #[cfg(test)]
+    fn parse_constraints(&mut self) -> Vec<Constraints> {
+        let mut constraints = Vec::<Constraints>::new();
+        constraints.push(self.parse_constraint());
+        while self.peek().tok == TOK_COMMA {
+            self.consume();
+            constraints.push(self.parse_constraint());
+        }
+        return constraints;
+    }
+
+    #[cfg(test)]
+    fn parse_constraint(&mut self) -> Constraints {
+        let t = self.peek();
+        let mut id = Vec::<Identifier>::new();
+        let _tvar = self.parse_identifier();
+        self.expect(TOK_COLON);
+        let identifier = self.parse_identifier();
+        id.push(identifier);
+        while self.peek().tok == TOK_ADD {
+            self.consume();
+            let identifier = self.parse_identifier();
+            id.push(identifier);
+        }
+        let con = Constraints {
+            base: self.base_node_from_token(&t),
```

Comment:
I think this should be `self.base_node_from_tokens(_tvar, end)` where `end` is the token representing the last identifier.